File: backend/services/emailService.js
const nodemailer = require('nodemailer');

// Création du transporteur SMTP
const transporter = nodemailer.createTransport({
    host: process.env.EMAIL_HOST,  // 👈 Utiliser host personnalisé
    port: process.env.EMAIL_PORT,  // 👈 Utiliser port personnalisé
    auth: {
      user: process.env.EMAIL_USER,
      pass: process.env.EMAIL_PASS
    }
  });

// Fonction générique pour envoyer un email
const sendEmail = async (to, subject, text) => {
  const mailOptions = {
    from: process.env.EMAIL_USER,
    to,
    subject,
    text
  };

  try {
    await transporter.sendMail(mailOptions);
    console.log(`✅ Email envoyé à ${to}`);
  } catch (error) {
    console.error('❌ Erreur envoi email:', error);
  }
};

// Fonction spécifique pour notifier du changement de statut
const sendTicketStatusUpdateEmail = async (userEmail, ticketId, newStatus) => {
  const subject = `Mise à jour du ticket #${ticketId}`;
  const text = `Bonjour,\n\nVotre ticket numéro #${ticketId} a été mis à jour.\n\nNouveau statut : ${newStatus}.\n\nMerci de votre confiance.`;
  
  await sendEmail(userEmail, subject, text);
};
// Fonction spécifique pour notifier un agent qu'il a été assigné à un ticket
const sendTicketAssignedEmail = async (agentEmail, ticketId , senderName, senderRole) => {
  const subject = `Nouveau ticket assigné #${ticketId}`;
  const text = `Bonjour,\n\nUn nouveau ticket vous a été assigné par ${senderName} (${senderRole}).\nNuméro du ticket : ${ticketId}.\n\nMerci de traiter cette demande rapidement.`;
  
  await sendEmail(agentEmail, subject, text);
};

module.exports = { sendEmail, sendTicketStatusUpdateEmail , sendTicketAssignedEmail };

File: backend-dot.net/Services/Email/EmailService.cs
using MailKit.Net.Smtp;
using MimeKit;
using Microsoft.Extensions.Configuration;

namespace SupportTicketApp.Services.Email
{
    public class EmailService : IEmailService
    {
        private readonly IConfiguration _config;

        public EmailService(IConfiguration config)
        {
            _config = config;
        }

        public async Task SendEmailAsync(string to, string subject, string body)
        {
            var message = new MimeMessage();
            message.From.Add(new MailboxAddress("Support", _config["Email:User"]));
            message.To.Add(MailboxAddress.Parse(to));
            message.Subject = subject;
            message.Body = new TextPart("plain") { Text = body };

            using var client = new SmtpClient();
            await client.ConnectAsync(_config["Email:Host"], int.Parse(_config["Email:Port"]), false);
            await client.AuthenticateAsync(_config["Email:User"], _config["Email:Pass"]);
            await client.SendAsync(message);
            await client.DisconnectAsync(true);
        }

        public Task SendTicketStatusUpdateEmail(string email, int ticketId, string status)
        {
            var subject = $"Mise à jour du ticket #{ticketId}";
            var text = $"Votre ticket #{ticketId} a été mis à jour.\nNouveau statut : {status}.";
            return SendEmailAsync(email, subject, text);
        }

        public Task SendTicketAssignedEmail(string email, int ticketId, string senderName, string senderRole)
        {
            var subject = $"Nouveau ticket assigné #{ticketId}";
            var text = $"Un ticket vous a été assigné par {senderName} ({senderRole}).";
            return SendEmailAsync(email, subject, text);
        }
    }
}
